### ISSUE TITLE
Typehint check fails for signatures which combine phpdoc and native typehints

### ISSUE THREAD

**Issue by mabar** [squizlabs/PHP_CodeSniffer#2503]
```php
/**
 * @param mixed $default
 * @return mixed
 */
function getRender(string $key, $default = NULL){}
```

`Unknown type hint "string" found for $default (Squiz.Commenting.FunctionComment.InvalidTypeHint)`

**Comment by jrfnl** [comment]
@mabar Have you tried adding a `@param` comment for `$key` ? The fact that that one is missing is what's confusing the sniff.

```php
/**
 * @param string $key
 * @param mixed  $default
 * @return mixed
 */
function getRender(string $key, $default = NULL){}
```

**Comment by mabar** [comment]
It should not depend at all `@param` defined in same order as in method signature because name of variable in doc is specified. Also it's useless to have typehint duplicated in phpdoc and in php (and against another sniff I use)

**Comment by jrfnl** [comment]
@mabar In that case, this may not be the right sniff for you to use. You could of course, silence the `Squiz.Commenting.FunctionComment.InvalidTypeHint` errorcode either in your custom ruleset or from the command-line..

**Comment by mabar** [comment]
Sniff fails in case of perfectly valid phpdoc and code. Not using it does not sounds like a solution to me.

**Comment by gsherwood** [comment]
> Sniff fails in case of perfectly valid phpdoc and code. Not using it does not sounds like a solution to me.

The sniff is working as designed. It clearly isn't working the way you want it to, but this sniff is not going to be modified to work in another way.

What PHPCS could really use is a collection of more generic doc sniffs based on today's common doc standards, but it's not something I personally have time for.

I'm going to close this because there isn't any wrong with the sniff itself.
